modules/vectra.py

```diff
@@ -2475,3 +2494,948 @@ def delete_account_note(self, account_id=None, note_id=None):
+        if "http://" in url: 
+            url = url.replace("http://", "https://")
+
+        return requests.get(url, headers=self.headers, params=params, verify=self.verify)
+
+    def get_all_detections(self, **kwargs):
+        """
+        Generator to retrieve all detections - all parameters are optional
+        :param c_score: certainty score (int) - will be removed with deprecation of v1 of api
+        :param c_score_gte: certainty score greater than or equal to (int) - will be removed with deprecation of v1 of api
+        :param category: detection category - will be removed with deprecation of v1 of api
+        :param certainty: certainty score (int)
+        :param certainty_gte: certainty score greater than or equal to (int)
+        :param detection: detection type
+        :param detection_type: detection type
+        :param detection_category: detection category
+        :param description:
+        :param fields: comma separated string of fields to be filtered and returned
+            possible values are: id, url, detection_url, category, detection, detection_category, 
+            detection_type, custom_detection, description, src_ip, state, t_score, c_score,
+            certainty, threat, first_timestamp, last_timestamp, targets_key_asset, 
+            is_targeting_key_asset, src_account, src_host, note, note_modified_by, 
+            note_modified_timestamp, sensor, sensor_name, tags, triage_rule_id, assigned_to, 
+            assigned_date, groups, is_marked_custom, is_custom_model
+        :param host_id: detection id (int)
+        :param is_targeting_key_asset: detection is targeting key asset (bool)
+        :param is_triaged: detection is triaged
+        :param last_timestamp: timestamp of last activity on detection (datetime)
+        :param max_id: maximum ID of detection returned
+        :param min_id: minimum ID of detection returned
+        :param ordering: field used to sort response
+        :param page: page number to return (int)
+        :param page_size: number of object to return in repsonse (int)
+        :param src_ip: source ip address of host attributed to detection
+        :param state: state of detection (active/inactive)
+        :param t_score: threat score (int) - will be removed with deprecation of v1 of api
+        :param t_score_gte: threat score is greater than or equal to (int) - will be removed with deprecation of v1 of api
+        :param tags: tags assigned to detection; this uses substring matching
+        :param targets_key_asset: detection targets key asset (bool) - will be removed with deprecation of v1 of api
+        :param threat: threat score (int)
+        :param threat_gte threat score is greater than or equal to (int)
+        :param note_modified_timestamp_gte: note last modified timestamp greater than or equal to (datetime)
+        """
+        resp = self._get_request(url='{url}/detections'.format(url=self.url), **self._generate_detection_params(kwargs))
```

Comment:
Is there a difference between the methods between VectraClientV2_2 and VectraClientV3?  For the methods, that the only difference is the auth (or decorators used), then it may be easier to consider using super() to invoke the parent class's method.

Reply:
Indeed not on all of them; I've adapted the code in consequence